pkg/services/serviceaccounts/manager/service.go

```diff
@@ -146,11 +153,42 @@ func (sa *ServiceAccountsService) Run(ctx context.Context) error {
 
 var _ serviceaccounts.Service = (*ServiceAccountsService)(nil)
 
-func (sa *ServiceAccountsService) CreateServiceAccount(ctx context.Context, orgID int64, saForm *serviceaccounts.CreateServiceAccountForm) (*serviceaccounts.ServiceAccountDTO, error) {
+func (sa *ServiceAccountsService) CreateServiceAccount(ctx context.Context, orgID int64, user *user.SignedInUser, saForm *serviceaccounts.CreateServiceAccountForm) (*serviceaccounts.ServiceAccountDTO, error) {
```

Comment:
Instead of having signed in users as a param you can extract it from context.
We set it here https://github.com/grafana/grafana/blob/main/pkg/services/contexthandler/contexthandler.go#L129 and then you can use `identity.GetRequester` to extract it

Reply:
Great suggestion, thanks! I'll refactor to use that.